testutil/base.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2014-2020 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package testutil
+
+import (
+	"gopkg.in/check.v1"
+)
+
+// BaseTest is a structure used as a base test suite for many of the pebble
+// tests.
+type BaseTest struct {
+	cleanupHandlers []func()
+}
```

Comment:
note for reviewers: this is the only reference to Pebble directly in the copied code. To keep the commit history clean, I am planning to first merge this PR and then update this docstring in a separate PR.